src/babashka/pods/sci.clj

```diff
@@ -19,6 +24,40 @@
               (string? var-value)
               (sci/eval-string* ctx var-value))))))
 
+(defn metadata-cache-file [pod-spec {:keys [:version]}]
+  (io/file (resolver/cache-dir {:pod/name pod-spec :pod/version version})
+           "metadata.cache"))
+
+(defn load-metadata-from-cache [pod-spec opts]
+  (let [cache-file (metadata-cache-file pod-spec opts)]
+    (when (.exists cache-file)
+      (with-open [r (PushbackInputStream. (io/input-stream cache-file))]
+        (impl/read r)))))
+
+(defn load-pod-metadata* [pod-spec {:keys [:version] :as opts}]
+  (let [metadata (impl/load-pod-metadata pod-spec opts)
+        cache-file (when (qualified-symbol? pod-spec) ; don't cache local pods b/c their namespaces can change
+                     (metadata-cache-file pod-spec opts))]
+    (when cache-file
+      (with-open [w (io/output-stream cache-file)]
+        (impl/write w metadata)))
+    metadata))
+
+(defn load-pod-metadata [ctx pod-spec {:keys [:version] :as opts}]
+  (let [metadata
+        (if-let [cached-metadata (when (qualified-symbol? pod-spec) ; don't cache local pods b/c their namespaces can change
+                                   (load-metadata-from-cache pod-spec opts))]
+          cached-metadata
+          (load-pod-metadata* pod-spec opts))]
+    (dorun
```

Comment:
Maybe we can just return the data instead of modifying the SCI atom? We can then save this data in a separate volatile or atom in babashka.main.